src/lightning/data/cache/config.py

```diff
@@ -0,0 +1,122 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+from lightning.data.cache.constants import INDEX_FILENAME
+from lightning.data.cache.downloader import get_downloader_cls
+from lightning.data.cache.pytree import treespec_loads
+from lightning.data.cache.sampler import ChunkedIndex
+
+
+class ChunksConfig:
+    def __init__(self, cache_dir: str, remote_dir: Optional[str]):
+        """The ChunksConfig reads the index files associated a chunked dataset and enables to map an index to its
+        chunk.
+
+        Arguments:
+            cache_dir: The path to cache folder.
+            remote_dir: The remote folder where the data are stored.
```

Comment:
can you specify the format here? E.g. s3 URI or just a regular filepath?

Reply:
done !